timemachine/potentials/bonded.py

```diff
@@ -231,5 +231,25 @@ def flat_bottom_bond(conf, params, box, bond_idxs):
 
     # compute energies
     k, r_min, r_max = params.T
-    bond_energies = (k / 4) * ((r > r_max) * ((r - r_max) ** 4) + (r < r_min) * ((r - r_min) ** 4))
-    return jnp.sum(bond_energies)
+    bond_nrgs = (k / 4) * ((r > r_max) * ((r - r_max) ** 4) + (r < r_min) * ((r - r_min) ** 4))
+    return bond_nrgs
+
+
+def flat_bottom_bond(conf, params, box, bond_idxs):
+    """
+    U(r; k, r_min, r_max) =
+        (k/4) * (r - r_max)**4 if r > r_max
+        (k/4) * (r - r_min)**4 if r < r_min
+    """
+    bond_nrgs = _flat_bottom_bond_impl(conf, params, box, bond_idxs)
+    return jnp.sum(bond_nrgs)
+
+
+def log_flat_bottom_bond(conf, params, box, bond_idxs, beta):
+    """
+    Implements the log inverse of the flat bottom potential
+    """
+    nrgs = _flat_bottom_bond_impl(conf, params, box, bond_idxs)
+    log_nrgs = -jnp.log(1 - jnp.exp(-beta * nrgs))
+    # note the extra 1/beta is to be consistent with the flat bottom bond
```

Comment:
Clarify?

(Is this to be consistent with some arbitrary convention, or is this to have units of energy?)

Reply:
Correct - we need the energies to have units of energy (so we can write the deselection term to also have a beta pre-pended to the log). This is also to ensure so that energies, forces, du/dps, etc. are all consistent with the other potentials.

Reply:
added a comment